.github/workflows/pypi-publish-test.yml

```diff
@@ -0,0 +1,50 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Publish to PyPi (Test)
+
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - reopened
+    branches:
+      - dev
```

Comment:
Should this be on pushes to dev? It might be better to have a "production" or "releases" branch to be the trigger as dev can have the possibility of becoming unstable

Reply:
Publishing to test may be unstable, which is ok, quote from Discord convo:

> The pipeline that PR introduces only runs when a PR targets `dev` and deploys it to the test index. In order for a deployment to the production index to happen, a workflow triggered by marking a new version runs. That process isn't done automatically, making it manual but very quick